packages/dara-core/dara/core/interactivity/any_variable.py

```diff
@@ -196,16 +196,22 @@ def is_valid(x):
 
         results = {}
 
+        saved_ws_channel = WS_CHANNEL.get()
+
         for session, channels in session_channels.items():
             for ws in channels:
+                # If we have a channel selected -- get just from that channel
+                if saved_ws_channel is not None and ws != saved_ws_channel:
```

Comment:
This condition should also be applied in the loop above where WS messages are fired, so we only request variables from the channel we care about

Reply:
I've moved it now to the for loop that assigns the variable we're looping on. I think it doesn't affect anything else except for getting these values, but do let me know if there is any problem with having them there! Cheers.